tests/jeos/grub2_gfxmode.pm

```diff
@@ -23,9 +23,13 @@ sub run {
         assert_script_run("sed -ie '/GFXPAYLOAD_LINUX=/s/=.*/=1024x768/' /etc/default/grub");
     }
 
+    # For Hyper-V and Xen PV we need to add special framebuffer provisions
     if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
         assert_script_run("sed -ie '/GRUB_CMDLINE_LINUX_DEFAULT=/s/\"\$/ video=hyperv_fb:1024x768\"/' /etc/default/grub");
     }
+    elsif (check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux')) {
+        assert_script_run("sed -ie '/GRUB_CMDLINE_LINUX_DEFAULT=/s/\"\$/ xenfb.video=4,1024,768\"/' /etc/default/grub");
+    }
```

Comment:
how do you like to extract the relevant part, e.g.
```perl
my $video;
if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
    $video = 'video=hyperv_fb:1024x768';
}
elsif (check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux')) {
    $video = 'xenfb.video=4,1024,768';
}
assert_script_run("sed -ie '/GRUB_CMDLINE_LINUX_DEFAULT=/s/\"\$/ ' . $video . '\"/' /etc/default/grub");
```
This way it might be more obvious what is the relevant part